### PR TITLE
chore: make python test dependencies optional

### PR DESCRIPTION
**Pull request by CollectiveUnicorn**
- Adds an optional `[dev]` dependency flag to the root `pyproject.toml` for test dependencies
- Closes: https://github.com/defenseunicorns/leapfrogai/issues/809

### CODE REVIEW

**Review by gphorvath — DISMISSED** [review]

**Comment by CollectiveUnicorn** [comment]
This PR affects and is affected by: https://github.com/defenseunicorns/leapfrogai/pull/808 - will wait on it to go through before updating this PR

**Review by jalling97 — APPROVED** [review]

**Review by justinthelaw — DISMISSED** [review]

File: src/leapfrogai_api/Makefile [hunk 1]
```diff
@@ -7,6 +7,7 @@ export SUPABASE_ANON_KEY=$(shell supabase status | grep -oP '(?<=anon key: ).*')
 install:
 	python -m pip install ../../src/leapfrogai_sdk
 	python -m pip install -e .
+	python -m pip install ../../[dev]
```
Comment (YrrepNoj) [inline src/leapfrogai_api/Makefile:10]: As written, this doesn't work with all shell types.
```suggestion
	python -m pip install "../../.[dev]"
```

**Review by YrrepNoj — CHANGES_REQUESTED** [review]

File: src/leapfrogai_api/Makefile (hunk 1 above)
Comment (CollectiveUnicorn) [reply src/leapfrogai_api/Makefile:10]: Good catch

**Review by YrrepNoj — DISMISSED** [review]

**Comment by gphorvath** [comment]
> This PR affects and is affected by: #808 - will wait on it to go through before updating this PR

It looks like #808 is going to need more work, so maybe retarget to main so it can be merged @CollectiveUnicorn?

**Comment by CollectiveUnicorn** [comment]
Right now its targeted at [port-load-tests](https://github.com/defenseunicorns/leapfrogai/tree/port-load-tests) since there's load test related dependencies included and readme changes.

**Review by gphorvath — APPROVED** [review]

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**